app/mailers/candidate_mailer.rb

```diff
@@ -282,7 +282,7 @@ def changed_offer(application_choice)
     @current_course_option = @application_choice.current_course_option
     @is_awaiting_decision = application_choice.self_and_siblings.decision_pending.any?
     @offers = @application_choice.self_and_siblings.select(&:offer?).map do |offer|
-      "#{offer.course_option.course.name_and_code} at #{offer.course_option.course.provider.name}"
+      "#{offer.current_course_option.course.name_and_code} at #{offer.current_course_option.course.provider.name}"
```

Comment:
or `ApplicationChoice#current_course`

Reply:
unfortunately I couldn't' use `current_course` because of the setup of most of the mailer tests, which are all `build_stubbed` which doesn't allow access to the AR methods. I had a bit of a go however it requires a rewrite to get this to work, so I am keen on keeping it this way, and I'm adding a ticket to revisit the mailers and audit/cleanup all of them in the sprint as we do need to update quite a few of those, where we can fix up the test setup. Let me know your thoughts